src/Network/Network/NetworkWatcher/ConnectionMonitor/NewAzureNetworkWatcherConnectionMonitorCommand.cs

```diff
@@ -57,64 +69,229 @@ public class NewAzureNetworkWatcherConnectionMonitorCommand : ConnectionMonitorB
             Mandatory = true,
             HelpMessage = "Location of the network watcher.",
             ParameterSetName = "SetByLocation")]
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "Location of the network watcher.",
+            ParameterSetName = "SetByLocationV2")]
         [LocationCompleter("Microsoft.Network/networkWatchers/connectionMonitors")]
         [ValidateNotNull]
         public string Location { get; set; }
 
         [Alias("ConnectionMonitorName")]
         [Parameter(
             Mandatory = true,
-            HelpMessage = "The connection monitor name.")]
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The connection monitor name.",
+            ParameterSetName = "SetByResource")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The connection monitor name.",
+            ParameterSetName = "SetByName")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The connection monitor name.",
+            ParameterSetName = "SetByLocation")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The connection monitor name.",
+            ParameterSetName = "SetByResourceV2")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The connection monitor name.",
+            ParameterSetName = "SetByNameV2")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The connection monitor name.",
+            ParameterSetName = "SetByLocationV2")]
         [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
         [Parameter(
-            Mandatory = true,
-            HelpMessage = "The ID of the connection monitor source.")]
+              Mandatory = true,
+              HelpMessage = "The ID of the resource used as the source by connection monitor.",
+              ParameterSetName = "SetByResource")]
+        [Parameter(
+              Mandatory = true,
+              HelpMessage = "The ID of the resource used as the source by connection monitor.",
+              ParameterSetName = "SetByName")]
+        [Parameter(
+              Mandatory = true,
+              HelpMessage = "The ID of the resource used as the source by connection monitor.",
+              ParameterSetName = "SetByLocation")]
         [ValidateNotNullOrEmpty]
         public string SourceResourceId { get; set; }
 
         [Parameter(
-            Mandatory = false,
-            HelpMessage = "Monitoring interval in seconds. Default value is 60 seconds.")]
+            Mandatory = true,
```

Comment:
This is a braking change - you cannot make this parameter mandatory without breaking existing scripts.

Reply:
fixed